github/provider_test.go

```diff
@@ -93,6 +93,55 @@ func TestAccProviderConfigure(t *testing.T) {
 
 	})
 
+	t.Run("can be configured with an app_auth block", func(t *testing.T) {
```

Comment:
I didn't have luck running those tests locally:

```shell
$ GITHUB_APP_ID='xxx' GITHUB_APP_INSTALLATION_ID='xxx' GITHUB_APP_PEM_FILE='xxx' GITHUB_OWNER='xxx' TF_ACC=1 go test -v ./... -run ^TestAccProviderConfigure

...
=== RUN   TestAccProviderConfigure/can_be_configured_to_run_anonymously
    provider_test.go:60: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: Duplicate provider configuration
        
          on terraform_plugin_test.tf line 3:
           3:                   provider "github" {}
        
        A default (non-aliased) provider configuration for "github" was already given
        at terraform_plugin_test.tf:1,1-18. If multiple configurations are required,
        set the "alias" argument for alternative configurations.
```

I don't quite understand where this is coming from and how to fix it.

Reply:
It's complaining because on line 111 below, you're already setting a provider, which is the provider defined [here](https://github.com/integrations/terraform-provider-github/blob/main/github/provider_test.go#L17-L19). 

If it makes you feel better, these tests are broken as-is, it's not you. I'm not sure if they ever even worked. 

I'm looking at the [Azure Terraform provider tests](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/provider/provider.go) as an example, and while they do have some configuration to tweak the provider, it's about tweaking the schema, not what's provided to it. 

I'm not sure what the best way of validating this behavior would be. [Hashicorp plugin development testing documentation](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing) doesn't appear to address this case. Probably in a best case scenario, we'd add another Actions workflow with a separate configuration to run tests as a whole. Thoughts?

Reply:
Thanks for the pointers!

> these tests are broken as-is, it's not you. I'm not sure if they ever even worked

I see that all acceptance tests are failing in CI. Do these tests run at all? Or are they just ignored?

Reading the docs, I think the `Config` field in the `TestStep`s is misused for the provider configuration (which as you pointed out is already initialized). Maybe it's possible to refactor this, but I wouldn't brave it (at least not in the scope of this PR).

> Probably in a best case scenario, we'd add another Actions workflow with a separate configuration to run tests as a whole. Thoughts?

I see that in the current CI setup there is a job per authentication method: anonymous, individual and organization. Are you suggesting adding another job for the app-based authentication? I think it would generally make sense. But that would require fixing current issues with CI and setting up an app for testing.